src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DynamicInterfaceCastableImplementation.Fixer.cs

```
@@ -0,0 +1,80 @@
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(context.Document);
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode enclosingNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            SyntaxNode declaration = generator.GetDeclaration(enclosingNode);
+            if (declaration == null || !CodeFixSupportsDeclaration(declaration))
+            {
+                return;
+            }
+            Diagnostic diagnostic = context.Diagnostics.First();
+            if (diagnostic.Id == DynamicInterfaceCastableImplementationAnalyzer.InterfaceMethodsMissingImplementationRuleId)
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        MicrosoftNetCoreAnalyzersResources.ImplementInterfacesOnDynamicCastableImplementation,
+                        async ct => await ImplementInterfacesOnDynamicCastableImplementation(declaration, context.Document, context.CancellationToken).ConfigureAwait(false),
+                        equivalenceKey: nameof(MicrosoftNetCoreAnalyzersResources.ImplementInterfacesOnDynamicCastableImplementation)),
+                    diagnostic);
+            }
+            else if (diagnostic.Id == DynamicInterfaceCastableImplementationAnalyzer.MethodsDeclaredOnImplementationTypeMustBeSealedRuleId)
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        MicrosoftNetCoreAnalyzersResources.SealMethodDeclaredOnImplementationType,
+                        async ct => await SealMemberDeclaredOnImplementationType(declaration, context.Document, context.CancellationToken).ConfigureAwait(false),
+                        equivalenceKey: nameof(MicrosoftNetCoreAnalyzersResources.SealMethodDeclaredOnImplementationType)),
+                    diagnostic);
+            }
```

else throw NotSupportedException or something that ensures the code path isn't reachable?

---

I think silently falling through here is ok unless the pattern in the repo otherwise is to throw.